### PR TITLE
Add find_renderer function to fix an error under macOS

### PR DESCRIPTION
**Pull request by andy1li**
Hi Phlya,

I love your library, it's really cool! However, when I tried to use it under the macOS, an error occurred saying that the renderer cannot be found. So I tried to add a find_renderer function to fix the problem and hope it can be merged into your wonderful library.

Regards,
Andy

### CODE REVIEW

**Comment by Phlya** [comment]
Thank you for your interest and kind words! And for the PR of course :)
I don't have a macOS computer - is the problem with macOS backend that there is no `get_renderer()` method, and it is just accessible as an attribute? I think I'll merge this PR, but it's also worth reporting this to matplotlib devs, this must be a bug... macOS backend is known to be weird.
Also, don't you think writing the function using `try`-`except` would be more pythonic?

**Comment by andy1li** [comment]
Yes, the exact error message was:
> AttributeError: 'FigureCanvasMac' object has no attribute 'get_renderer'

Googling the error message yielded similar problems with matplotlib on macOS, such as [this one](http://stackoverflow.com/questions/13216520/mac-osx-attributeerror-figurecanvasmac-object-has-no-attribute-restore-reg).

And [this post](http://stackoverflow.com/questions/22667224/matplotlib-get-text-bounding-box-independent-of-backend) pointed me in the right direction of fixing the problem (although it did not solve it directly)

Finally, after printing out the `canvas.__dict__` from your library, it seems that the mac backend only has a `renderer `attribute.

Yes, you are definitely right! Using try and except would be more pythonic. (I was too occupied with getting it to work...)

Thanks again for your great library! The plots look so much better after `adjust_text`!

**Comment by andy1li** [comment]
Don't know whether the comment above opened the PR again or what. Since it's open, might as well update the code.

File: adjustText/adjustText.py [hunk 1]
```diff
@@ -19,20 +19,18 @@ def get_midpoint(bbox):
...
 def get_points_inside_bbox(x, y, bbox):
     x1, y1, x2, y2 = bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax
     x_in = np.logical_and(x>x1, x<x2)
     y_in = np.logical_and(y>y1, y<y2)
     return np.where(x_in & y_in)[0]
 
+def get_renderer(fig):
+    try: 
+        return fig.canvas.get_renderer()
+    except:
```
Comment (Phlya) [inline adjustText/adjustText.py]: `except AttributeError:` is preferable, since we don't know what other kinds of errors there might be raised.

**Comment by Phlya** [comment]
Thank you, merged!

**Comment by andy1li** [comment]
Does it have a delay or something? Nothing seems to have happened and it says closed (not merged) actually...

**Comment by Phlya** [comment]
Sorry, forgot to confirm the merge!

**Pull request merged**